### DISCUSSION TITLE
Can't convince llama.cpp to read and write files

### DISCUSSION THREAD

**Discussion by varun867** [ggml-org/llama.cpp#1510] [Q&A] [answered]
I have llama.cpp working locally on a windows machine.  I am running ggml-vic7b-uncensored-q5_0.bin and ggml-vic13b-uncensored-q5_0.bin

**some times i want to give it input like**
Can you read the file D:\test\SP.sql and explain to me what is happening in there ? 
Can you read the article in the location D:\Articles\article1.txt and generate flashcards  in csv format and write it to D:\Flashcards\article1.csv

Is this even possible? How can i do this?  it simply says it cannot read from path even when i run it as Administrator

**Comment by BetaDoggo** [comment, marked as answer]
On its own llama is a text model that can do nothing but generate text. What you're looking for is probably something like [auto-GPT](https://github.com/Significant-Gravitas/Auto-GPT). Be warned that these types of applications don't work very well even with the gpt4 model, so using llama with it probably won't accomplish much. If you really want to try it some UIs like [text-generation-webui](https://github.com/oobabooga/text-generation-webui) can emulate the OpenAI API so that would be a good place to start.

**Comment by RahulVivekNair** [comment]
Llama.cpp can only run language models. It's sole purpose is to generate text and do no have access to your computer's file system. There are projects which uses the language model's reasoning skills to do what you're asking and expose the user file system to the program such as langchain and autogpts.

**Comment by varun867** [comment]
Thank you for your suggestions. I will research more in that direction
